crates/bench/src/game.rs

```diff
@@ -0,0 +1,198 @@
+/// Load test to see large variations on performance
+///
+/// NOTE: It should be running with `--release` or `--profile bench` when looking with perf tools...
+use criterion::black_box;
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+use tempdir::TempDir;
+
+use spacetimedb::db::relational_db::{open_db, RelationalDB};
+use spacetimedb::error::DBError;
+use spacetimedb::execution_context::ExecutionContext;
+use spacetimedb::sql::execute::run;
+use spacetimedb::subscription::subscription::create_table;
+use spacetimedb_lib::error::ResultTest;
+use spacetimedb_lib::identity::AuthCtx;
+use spacetimedb_primitives::TableId;
+use spacetimedb_sats::{product, AlgebraicType};
+
+fn make_test_db() -> Result<(RelationalDB, TempDir), DBError> {
+    let tmp_dir = TempDir::new("stdb_test")?;
+    let stdb = open_db(&tmp_dir, true, false)?.with_row_count(Arc::new(|_, _| i64::MAX));
+    Ok((stdb, tmp_dir))
+}
+
+#[derive(Clone, Copy)]
+pub struct GameData {
+    location_state: u64,
+    footprint_tile_state: u64,
+}
+
+impl GameData {
+    /// Will generate on `release`:
+    /// Chunks * 100
+    /// FootprintTileState * 1'000
+    /// LocationState * 1'000.000 rows
+    pub fn new(base_rows: u64) -> Self {
+        Self {
+            footprint_tile_state: base_rows,
+            location_state: base_rows * 10,
+        }
+    }
+}
+
+#[cfg(not(debug_assertions))]
+pub const BASE_ROWS: usize = 1_000;
+#[cfg(debug_assertions)]
+pub const BASE_ROWS: usize = 10;
+const EXPECT_ROWS: usize = BASE_ROWS * 1_000;
```

Comment:
Lets keep this simple for right now and just have a single table size. Let's do 1M rows each for both tables.